### DISCUSSION TITLE
How to run integration tests for BPMNs that include service task (workers)

### DISCUSSION THREAD

**Discussion by seattle-classmate** [camunda-community-hub/community#149] [Q&A]
My team is using Camunda 8 in production and we are planning to add integration tests.
We want to achieve the following integration test functionality:
- Test deployment of BPMN to zeebe
- Test the start and completion of BPMN workflows that include service tasks (validate the execution of database calls to a dockerized datastore as part of the worker implementation)

I am using `zeebe-process-test-extension` maven dependency to spin up an embedded zeebe engine i.e. @ZeebeProcessTest and its assertions
The BPMN to be tested has a service task that queries a documentdb datastore that is running in docker.
I added this code to bootstrap the worker - 

`final var activateJobsResponse = zeebeClient.newActivateJobsCommand().jobType(jobType).maxJobsToActivate(count).send().join();
`
and passed in the jobType as per the worker implementation in src/main/java folder.

After running the test, the embedded zeebe engine comes up fine, the BPMN gets deployed correctly and the process instance starts but instead of trigger the above mentioned worker implementation, it triggers a new worker and asserts the output of the BPMN to false.
Which means, it did not pick the correct worker instead ran a new worker of the same jobType.

How can I resolve this and make sure the test picks the worker inside src/main/java folder?

Note - I do not want to mock the worker implementation

**Comment by saig0** [comment]
Can you share your code?

In general, you could start your job worker inside the test case using the provided Zeebe client. 

If you're using Spring, check the spring-zeebe documentation [here](https://github.com/camunda-community-hub/spring-zeebe?tab=readme-ov-file#writing-test-cases).
